### PR TITLE
nodePackages.ollama: init at 0.5.0

### LINKED ISSUE #298315: Package request: ollama-js
**Project description**

Package request: nodePackages.ollama

**Metadata**

* homepage URL: https://github.com/ollama/ollama-js/
* source URL: https://github.com/NixOS/nixpkgs/blob/nixos-unstable/pkgs/development/node-packages/node-packages.nix
* license: MIT
* platforms: unix, linux, darwin

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by Rusted-Silver**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your contribution.

1. Read the contributor guide (https://nix.dev/contributing/how-to-contribute.html) to learn how to format the PR title and commit log message
2. I see that the `node-packages.nix` contains many package updates, is it intended ?

File: pkgs/development/node-packages/node-packages.nix [hunk 1]
```diff
@@ -34375,7 +34375,7 @@ let
       version = "2.5.2";
       src = fetchurl {
         url = "https://registry.npmjs.org/jsesc/-/jsesc-2.5.2.tgz";
-        sha512 = "OYu7XEzjkCQ3C5Ps3QIZsQfNpqoJyZZA99wd9aWd05NCtC5pWOkShK2mkL6HXQR6/Cy2lbNdPlZBpuQHXE63gA==";
+        sha512 = /OYu7XEzjkCQ3C5Ps3QIZsQfNpqoJyZZA99wd9aWd05NCtC5pWOkShK2mkL6HXQR6/Cy2lbNdPlZBpuQHXE63gA==";
```
Comment (drupol) [inline pkgs/development/node-packages/node-packages.nix:34378]: I don't think this is valid...

Comment (Kahlenden) [reply pkgs/development/node-packages/node-packages.nix:34378]: yea i noticed, just a silly little mistake : D

Comment (Kahlenden) [reply pkgs/development/node-packages/node-packages.nix:34378]: that's why i want to run the autogenerate instead, my sleepy head is not great to make manual changes

Comment (Kahlenden) [reply pkgs/development/node-packages/node-packages.nix]: I can't believe i made a mistake twice, I'm so sorry...

**Comment by abysssol** [comment]
What's the status of this pr? Are there any remaining blockers? From what I can see, it looks ready to merge.

**Comment by SuperSandro2000** [comment]
We don't want to add new packages to nodePackages and it should be packaged separately. You seem to care for this PR. Could you open a PR for that?

**Comment by SuperSandro2000** [comment]
> Why? Could you point me to any relevant documentation/issues/etc?

see #229475

**Comment by abysssol** [comment]
@SuperSandro2000 From what I can tell, ollama-js is just a library, which makes me wonder whether it should be packaged at all. From what I can tell, all node packages have a cli or some interactive artifact; I'm not even sure how js libraries packaged in nix can be used, or if they can at all. What do you think?

@Kahlenden Why do you want ollama-js to be packaged in nixpkgs? What use case does that enable for you, in what ways can it be used?

**Comment by SuperSandro2000** [comment]
Good question. I do not know either. It could be that some program required it to be build or so but then it should be packaged there.

Closing

**Pull request closed without merging**